.github/workflows/go-tests.yml

```diff
@@ -0,0 +1,125 @@
+name: go-tests
+on:
+  pull_request:
+  push:
+    branches: [main]
+
+jobs:
+  golangci:
+    name: golangci-lint
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c # pin@v3.3.0
+      - uses: actions/setup-go@6edd4406fa81c3da01a34fa6f6343087c207a568 # pin@v3.5.0
+      - name: golangci-lint
+        uses: golangci/golangci-lint-action@08e2f20817b15149a52b5b3ebe7de50aff2ba8c5  # pin@v3.4.0
+        with:
+          version: v1.51.2
+          args: |
+            --disable-all \
+            --timeout 10m \
+            --enable gofmt \
+            --enable gosimple \
+            --enable govet \
+            --verbose
+          skip-pkg-cache: true
+          skip-build-cache: true
+      - uses: hashicorp/actions-slack-status@v1
+        if: failure() && github.ref == 'refs/heads/main'
```

Comment:
Do we only want to report failures on `main` and not on other branches (ie in pull requests)?